### PR TITLE
#172 - Added integration test for caching proxy

### PR DESCRIPTION
**Pull request by olegmoz**
Part of #172 
Added target integration test for caching proxy.

### CODE REVIEW

File: src/test/java/com/artipie/docker/http/CachingProxyITCase.java [hunk 1]
```diff
@@ -100,4 +124,45 @@ void shouldPull() throws Exception {
...
+        this.cli.run("pull", image);
+        this.awaitManifestCached(digest);
+        this.cli.run("image", "rm", image);
+        this.client.stop();
+        final String output = this.cli.run("pull", image);
+        MatcherAssert.assertThat(
+            output,
+            new StringContains(
+                false,
+                String.format("Status: Downloaded newer image for %s", image)
```
Comment (olenagerasimova) [inline src/test/java/com/artipie/docker/http/CachingProxyITCase.java:146]: @olegmoz I'd suggest extracting this expected message into constant: if it changes there will be only one place to correct

File: src/test/java/com/artipie/docker/http/CachingProxyITCase.java [hunk 2]
```diff
@@ -100,4 +124,45 @@ void shouldPull() throws Exception {
...
+            )
+        );
+    }
+
+    private void awaitManifestCached(final String digest) throws InterruptedException {
+        final Manifests manifests = this.cache.repo(new RepoName.Simple("dotnet/core/runtime"))
+            .manifests();
+        final ManifestRef ref = new ManifestRef.FromDigest(new Digest.FromString(digest));
+        final Stopwatch stopwatch = Stopwatch.createStarted();
+        while (manifests.get(ref).toCompletableFuture().join().isEmpty()) {
```
Comment (olenagerasimova) [inline src/test/java/com/artipie/docker/http/CachingProxyITCase.java:156]: @olegmoz completable future has `public T get(long timeout, java.util.concurrent.TimeUnit unit)` method to wait most the given time to complete, seems like you could use it here

File: src/test/java/com/artipie/docker/http/CachingProxyITCase.java (hunk 1 above)
Comment (olegmoz) [reply src/test/java/com/artipie/docker/http/CachingProxyITCase.java]: @olenagerasimova solved it by introducing a method for building matcher

File: src/test/java/com/artipie/docker/http/CachingProxyITCase.java (hunk 2 above)
Comment (olegmoz) [reply src/test/java/com/artipie/docker/http/CachingProxyITCase.java]: @olenagerasimova that's funny how I forgot about it, fixed it now, thanks!

**Pull request merged**